commons-test/src/main/java/org/infinispan/commons/test/PolarionJUnitXMLReporter.java

```diff
@@ -0,0 +1,333 @@
+
+      if ((ITestResult.FAILURE == tr.getStatus()) || (ITestResult.SKIP == tr.getStatus())) {
+         doc.push(XMLConstants.TESTCASE, attrs);
+
+         if (ITestResult.FAILURE == tr.getStatus()) {
+            createFailureElement(doc, tr);
+         } else if (ITestResult.SKIP == tr.getStatus()) {
+            createSkipElement(doc, tr);
+         }
+
+         doc.pop();
+      } else {
+         doc.addEmptyElement(XMLConstants.TESTCASE, attrs);
+      }
+   }
+
+   private void createFailureElement(XMLStringBuffer doc, ITestResult tr) {
+      Properties attrs = new Properties();
+      Throwable t = tr.getThrowable();
+      if (t != null) {
+         attrs.setProperty(XMLConstants.ATTR_TYPE, t.getClass().getName());
+         String message = t.getMessage();
+         if ((message != null) && (message.length() > 0)) {
+            attrs.setProperty(XMLConstants.ATTR_MESSAGE, encodeAttr(message)); // ENCODE
+         }
+         doc.push(XMLConstants.FAILURE, attrs);
+         doc.addCDATA(Utils.stackTrace(t, false)[0]);
+         doc.pop();
+      } else {
+         doc.addEmptyElement(XMLConstants.FAILURE); // THIS IS AN ERROR
+      }
+   }
+
+   private void createSkipElement(XMLStringBuffer doc, ITestResult tr) {
+      doc.addEmptyElement("skipped");
+   }
+
+   private String encodeAttr(String attr) {
+      String result = replaceAmpersand(attr, ENTITY);
+      for (Map.Entry<String, Pattern> e : ATTR_ESCAPES.entrySet()) {
+         result = e.getValue().matcher(result).replaceAll(e.getKey());
+      }
+
+      return result;
+   }
+
+   private String replaceAmpersand(String str, Pattern pattern) {
+      int start = 0;
+      int idx = str.indexOf('&', start);
+      if (idx == -1) {
+         return str;
+      }
+      StringBuilder result = new StringBuilder();
+      while (idx != -1) {
+         result.append(str.substring(start, idx));
+         if (pattern.matcher(str.substring(idx)).matches()) {
+            // do nothing it is an entity;
+            result.append("&");
+         } else {
+            result.append("&amp;");
+         }
+         start = idx + 1;
+         idx = str.indexOf('&', start);
+      }
+      result.append(str.substring(start));
+
+      return result.toString();
+   }
+
+   /**
+    * Reset all member variables for next test.
+    */
+   private void resetAll() {
+      m_allTests = Collections.synchronizedList(Lists.<ITestResult>newArrayList());
+      m_configIssues = Collections.synchronizedList(Lists.<ITestResult>newArrayList());
+      m_numFailed.set(0);
+      m_numSkipped.set(0);
+   }
+
+   private String generateFileName(ISuite suite) {
+      ITestNGMethod[] testMethods = suite.getResults().values().iterator().next().getTestContext().getAllTestMethods();
+      if (suite.getName().contains("Surefire suite") && testMethods.length > 0) {
+         return "POLARION-" + testMethods[0].getConstructorOrMethod().getDeclaringClass().getPackage().getName();
```

Comment:
I need to work on the file naming code a little more. Picking the package of the first method doesn't seem to always be a good choice. :-(